### PR TITLE
Importing ReflTBL will now load it into processing table in one go

### LINKED ISSUE #14557: Importing a TBL file created from the v1 Refl GUI to the v2 gui should be easier
Version 1 of the GUI (Max's) works of a TBL file. These can be saved and loaded and effectively remember the entries in the table.

In Version 2 of the GUI, we work off TableWorkspaces. However we also introduced the ability to import a TBL file from the menu. However the import does not fully complete the import. You have to do an import as one step and a load as another. We should just be able to Import the table in one-shot.
screen shot 2015-11-24 at 10 04 49

### PR DESCRIPTION
**Pull request by MattKing06**
Previously users had to import the TBL file, which loaded it into the ADS, and then open the table so that it was present in the processing table. Now importing a TBL file will load the table into the ADS and the processing table in one go.

**Code Review**:
- This is currently done based on the string containing the algorithm name in `showAlgorithmDialog`
- if the the algorithm is `LoadReflTBL` then we bring in the loading functionality
- this `showAlgorithmDialog` function is used elsewhere with different algorithms in the MainView
- Could we possible introduce a second parameter to the method called `AdditionalCode` which will contain a string of code specific to running the `LoadReflTBL` algorithm?

**To Test**:
- Data can be found in `SampleData-ISIS`: the TBL file is `INTER_NR_test2.tbl`
- Go to Interfaces > Reflectometry > ISIS Reflectometry (Polref)
- Once the interface has loaded, go to `Reflectometry` on the menu bar
- Click `Import .TBL`
- Navigate to the `INTER_NR_test2.tbl` and give it an OutputWorkspace name
- Click `Load`
- Ensure the table is loaded straight into the Processing table (table on the right of the interface)

**Release Notes**:
http://www.mantidproject.org/Release_Notes_3_6_Reflectometry#Reflectometry_Reduction_Interface

Fixes #14557

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp [hunk 1]
```diff
@@ -438,10 +438,19 @@ Show the user the dialog for an algorithm
 void QtReflMainView::showAlgorithmDialog(const std::string &algorithm) {
   std::stringstream pythonSrc;
   pythonSrc << "try:\n";
-  pythonSrc << "  " << algorithm << "Dialog()\n";
+  pythonSrc << "  algm = " << algorithm << "Dialog()\n";
+  if (algorithm == "LoadReflTBL") {
```
Comment (OwenArnold) [inline MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp]: 1. Split this functionality into a new public method called showImportDialog.
2. Don't do the algorithm check. Instead, look at the algorithm properties via getProperties. Look for a key called 'OutputWorkspace' if it exists. Fetch it.

File: MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflMainView.h [hunk 2]
```diff
@@ -11,84 +11,89 @@
...
+  // Dialog/Prompt methods
+  virtual std::string askUserString(const std::string &prompt,
+                                    const std::string &title,
+                                    const std::string &defaultValue) = 0;
+  virtual bool askUserYesNo(std::string prompt, std::string title) = 0;
+  virtual void giveUserInfo(std::string prompt, std::string title) = 0;
+  virtual void giveUserWarning(std::string prompt, std::string title) = 0;
+  virtual void giveUserCritical(std::string prompt, std::string title) = 0;
+  virtual void showAlgorithmDialog(const std::string &algorithm) = 0;
+  virtual void showImportDialog() = 0;
```
Comment (OwenArnold) [inline MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/Reflectometry/ReflMainView.h]: In future, use the override keyword when declaring new pure virtual functions please

File: MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp [hunk 3]
```diff
@@ -438,10 +438,26 @@ Show the user the dialog for an algorithm
...
+  pythonSrc << "  algm = "
+            << "LoadReflTBL"
+            << "Dialog()\n";
+  pythonSrc << "  print algm.getPropertyValue(\"OutputWorkspace\")\n";
+  pythonSrc << "except:\n";
+  pythonSrc << "  pass\n";
+  // outputWorkspaceName will hold the name of the workspace
+  // otherwise this should be an empty string.
+  QString outputWorkspaceName =
+      runPythonCode(QString::fromStdString(pythonSrc.str()), false);
```
Comment (OwenArnold) [inline MantidQt/CustomInterfaces/src/Reflectometry/QtReflMainView.cpp]: Could have used a comment here explaining what was happening with setModel.

File: MantidQt/CustomInterfaces/test/ReflMainViewPresenterTest.h [hunk 4]
```diff
@@ -1531,7 +1531,7 @@ class ReflMainViewPresenterTest : public CxxTest::TestSuite {
     NiceMock<MockView> mockView;
     NiceMock<MockProgressableView> mockProgress;
     ReflMainViewPresenter presenter(&mockView, &mockProgress);
-    EXPECT_CALL(mockView, showAlgorithmDialog("LoadReflTBL"));
+    EXPECT_CALL(mockView, showImportDialog());
```
Comment (OwenArnold) [inline MantidQt/CustomInterfaces/test/ReflMainViewPresenterTest.h]: Good.

**Comment by OwenArnold** [comment]
@MattKing06 Please add something about this to the reflectometry release notes.

**Comment by MattKing06** [comment]
@OwenArnold I've added the changes to the release notes: http://www.mantidproject.org/Release_Notes_3_6_Reflectometry#Reflectometry_Reduction_Interface

**Pull request merged**